crates/red_knot_python_semantic/src/types.rs

```diff
@@ -133,6 +139,21 @@ fn symbol<'db>(db: &'db dyn Db, scope: ScopeId<'db>, name: &str) -> Symbol<'db>
     {
         return Symbol::Type(Type::BooleanLiteral(true), Boundness::Bound);
     }
+    if name == "platform"
+        && file_to_module(db, scope.file(db)).is_some_and(|module| module.name() == "sys")
```

Comment:
What if it's a random user-defined module called `sys`? Do we have tests for that?

What you have here _might_ actually be okay for `sys` specifically, because of the fact that certain core stdlib modules can't be overridden by first-party or third-party modules (Python's import machinery _always_ gives priority to the stdlib for these modules), and `sys` is one of them. But we should at least add a comment to this branch, if that's the reason why we're not checking the search path of this module.

I think the first branch of this function is definitely incorrect, actually, now that I look at it again: `typing` is _not_ one of the modules that is immune from being overridden by first-party code. So we need to check that `module.search_path().is_standard_library()` in the first branch of this function before applying the special-casing for `typing.TYPE_CHECKING`